depends/patches/qt/static_fixes.patch

```diff
@@ -0,0 +1,80 @@
+index 26b9bf89633..0c546d09a8b 100644
+--- a/qtbase/cmake/3rdparty/extra-cmake-modules/find-modules/FindXCB.cmake
++++ b/qtbase/cmake/3rdparty/extra-cmake-modules/find-modules/FindXCB.cmake
+@@ -145,7 +145,7 @@ endforeach()
+ set(XCB_XCB_component_deps)
+ set(XCB_COMPOSITE_component_deps XCB XFIXES)
+ set(XCB_DAMAGE_component_deps XCB XFIXES)
+-set(XCB_IMAGE_component_deps XCB SHM)
++set(XCB_IMAGE_component_deps XCB SHM AUX)
+ set(XCB_RENDERUTIL_component_deps XCB RENDER)
+ set(XCB_XFIXES_component_deps XCB RENDER SHAPE)
+ set(XCB_XVMC_component_deps XCB XV)
+diff --git a/src/gui/configure.cmake b/src/gui/configure.cmake
+index 99c517e3581..a2e644f77d9 100644
+--- a/qtbase/src/gui/configure.cmake
++++ b/qtbase/src/gui/configure.cmake
+@@ -80,10 +80,14 @@ if((X11_SUPPORTED) OR QT_FIND_ALL_PACKAGES_ALWAYS)
+     qt_find_package(XCB 0.3.9 COMPONENTS ICCCM PROVIDED_TARGETS XCB::ICCCM MODULE_NAME gui QMAKE_LIB xcb_icccm)
+ endif()
+ qt_add_qmake_lib_dependency(xcb_icccm xcb)
++if((X11_SUPPORTED) OR QT_FIND_ALL_PACKAGES_ALWAYS)
++    qt_find_package(XCB 0.3.8 COMPONENTS UTIL PROVIDED_TARGETS XCB::UTIL MODULE_NAME gui QMAKE_LIB xcb_util)
++endif()
++qt_add_qmake_lib_dependency(xcb_util xcb)
+ if((X11_SUPPORTED) OR QT_FIND_ALL_PACKAGES_ALWAYS)
+     qt_find_package(XCB 0.3.9 COMPONENTS IMAGE PROVIDED_TARGETS XCB::IMAGE MODULE_NAME gui QMAKE_LIB xcb_image)
+ endif()
+-qt_add_qmake_lib_dependency(xcb_image xcb_shm xcb)
++qt_add_qmake_lib_dependency(xcb_image xcb_shm xcb_util xcb)
+ if((X11_SUPPORTED) OR QT_FIND_ALL_PACKAGES_ALWAYS)
+     qt_find_package(XCB 0.3.9 COMPONENTS KEYSYMS PROVIDED_TARGETS XCB::KEYSYMS MODULE_NAME gui QMAKE_LIB xcb_keysyms)
+ endif()
+@@ -488,6 +492,7 @@ qt_config_compile_test(xcb_syslibs
+     LIBRARIES
+         XCB::CURSOR
+         XCB::ICCCM
++        XCB::UTIL
+         XCB::IMAGE
+         XCB::KEYSYMS
+         XCB::RANDR
+@@ -503,6 +508,7 @@ qt_config_compile_test(xcb_syslibs
+ "// xkb.h is using a variable called 'explicit', which is a reserved keyword in C++
+ #define explicit dont_use_cxx_explicit
+ #include <xcb/xcb.h>
++#include <xcb/xcb_util.h>
+ #include <xcb/xcb_image.h>
+ #include <xcb/xcb_keysyms.h>
+ #include <xcb/xcb_cursor.h>
+diff --git a/src/plugins/platforms/xcb/CMakeLists.txt b/src/plugins/platforms/xcb/CMakeLists.txt
+index e8fb442dd43..e964138115c 100644
+--- a/qtbase/src/plugins/platforms/xcb/CMakeLists.txt
++++ b/qtbase/src/plugins/platforms/xcb/CMakeLists.txt
+@@ -52,6 +52,7 @@ qt_internal_add_module(XcbQpaPrivate
+         Qt::GuiPrivate
+         XCB::CURSOR
+         XCB::ICCCM
++        XCB::UTIL
+         XCB::IMAGE
+         XCB::KEYSYMS
+         XCB::RANDR
+--- a/qtbase/src/gui/configure.cmake
++++ b/qtbase/src/gui/configure.cmake
+@@ -499,6 +499,7 @@ qt_config_compile_test(xcb_syslibs
+         XCB::XFIXES
+         XCB::XKB
+         XCB::XCB
++        X11::Xau
```

Comment:
I cannot find this change upstream, but it is correct. Otherwise, the `xcb_syslibs` test fails with the following errors:
```
xcb_auth.c:(.text+0x81): undefined reference to `XauGetBestAuthByAddr'
...
xcb_auth.c:(.text+0x292): undefined reference to `XauDisposeAuth'
```

Reply:
There's a related report to libxcb here: https://gitlab.freedesktop.org/xorg/lib/libxcb/-/merge_requests/25.